scipy/signal/filter_design.py

```diff
@@ -3504,6 +3504,196 @@ def besselap(N, norm='phase'):
     return asarray([]), asarray(p, dtype=complex), float(k)
 
 
+def iirnotch(w0, bw, ab=3):
+    """
+    Design second-order IIR notch digital filter.
+
+    Parameters
+    ----------
+    w0 : float
+        A scalar that must satisfy  0 < w0 < 1. For notch filters,
+        w0 is the normalized frequency to remove from the filtered
+        signal (w0 = 1 corresponds to half of the sample frequency).
+    bw : float
+        Filter bandwidth.
+    ab : float, optional
+        Bandwidht maximum atenuation in decibels. By default, `ab`
+        is 3, so the atenuation is -3 dB.
+
+    Returns
+    -------
+    b, a : ndarray, ndarray
+        Numerator (`b`) and denominator (`a`) polynomials
+        of the IIR filter.
+
+    Notes
+    -----
+    The quality factor (Q factor) for the filter is related to the
+    filter bandwidth by Q = w0/bw (for the default value of `ab`).
+
+    .. versionadded: 0.19.0
+
+    References
+    ----------
+    .. [1] Sophocles J. Orfanidis, "Introduction To Signal Processing",
+           Prentice-Hall, 1996
+
+    Examples
+    --------
+    Design and plot filter to remove the 60Hz component from a
+    signal sampled at 1000Hz, using a quality factor Q=30
+
+    >>> from scipy import signal
+    >>> import matplotlib.pyplot as plt
+
+    >>> fs = 1000.0  # Sample frequency (Hz)
+    >>> f0 = 60.0  # Frequency to be removed from signal (Hz)
+    >>> q = 30.0  # Quality factor
+    >>> w0 = f0/(fs/2) # Normalized Frequency
+    >>> bw = w0/q # Bandwidth
+    >>> # Design notch filter
+    >>> b, a = signal.iirnotch(w0, bw)
+
+    >>> # Frequency response
+    >>> w, h = signal.freqz(b, a, 2**14)
```

Comment:
The notch filter is a band-stop filter with very narrow bandwidth. For the exemple the magnitude of the filter frequency response will decrease very fast to zero around 60Hz.
If to few points are sampled from the frequency response the plot will not represent the frequency response properly. For instance, the graph below was generated for 2 ** 9 points.

![screenshot 2016-07-23 11 23 30](https://cloud.githubusercontent.com/assets/16557411/17078373/0ab1c5a0-50c8-11e6-8734-4688ac156bc4.png)